manifests/deployment.yaml

```diff
@@ -0,0 +1,24 @@
+---
+kind: Deployment
+apiVersion: apps/v1
+metadata:
+  name: objectstorage-controller
+  namespace: objectstorage-system
+spec:
+  replicas: 1
```

Comment:
Does the controller have some kind of leader election? If not, I think it's easier to simply deploy this as a `StatefulSet` of a single replica.
Keep in mind a `Deployment` with `replicas: 1` does not *guarantee* only a single replica is running at any point in time.